extmod/uasyncio/stream.py

```diff
@@ -123,7 +123,11 @@ def close(self):
         self.task.cancel()
 
     async def wait_closed(self):
-        await self.task
+        try:
+            await self.task
+        except core.CancelledError:
```

Comment:
Does this now unintentionally suppress cancellations of tasks that wait on the server? Eg:
```python
async def task():
    srv = asyncio.start_server(...)
    srv.close()
    await srv.wait_closed()

async def main():
    t = asyncio.create_task(task())
    asyncio.sleep(0)
    t.cancel() # CancelledError will be suppressed?
```

Reply:
I think the sleep version is the way to go. I've pushed that and updated the tests with a case that was broken with the initial version of this PR based on your suggestion above.